src/main/java/core/model/player/Player.java

```diff
@@ -2194,54 +2223,69 @@ public void setFutureTraining(FuturePlayerTraining.Priority prio, HODateTime fro
-     *  if there are more than one selected priorities, "individual priorities" is returned
-     *  if is no user selected priority, the best position information is returned
+     * if there are more than one selected priorities, "individual priorities" is returned
+     * if is no user selected priority, the best position information is returned
      */
     public String getTrainingPriorityInformation(HODateTime nextWeek) {
-        String ret=null;
-        for ( var t : getFuturePlayerTrainings()) {
+        String ret = null;
+        for (var t : getFuturePlayerTrainings()) {
             //
-            if ( !t.endsBefore(nextWeek)){
-                if ( ret != null ){
+            if (!t.endsBefore(nextWeek)) {
+                if (ret != null) {
                     ret = HOVerwaltung.instance().getLanguageString("trainpre.individual.prios");
                     break;
                 }
                 ret = t.getPriority().toString();
             }
         }
-        if ( ret != null ) return ret;
+        if (ret != null) return ret;
         return getBestPositionInfo(null, false);
 
     }
 
-    private static int[] trainingSkills= { KEEPER, SET_PIECES, DEFENDING, SCORING, WINGER, PASSING, PLAYMAKING };
+    private static int[] trainingSkills = {KEEPER, SET_PIECES, DEFENDING, SCORING, WINGER, PASSING, PLAYMAKING};
 
     /**
      * Calculates skill status of the player
      *
-     * @param previousID Id of the previous download. Previous player status is loaded by this id.
+     * @param previousID    Id of the previous download. Previous player status is loaded by this id.
      * @param trainingWeeks List of training week information
      */
     public void calcSubskills(int previousID, List<TrainingPerWeek> trainingWeeks) {
-        var playerBefore = DBManager.instance().getSpieler(previousID).stream()
-                .filter(i -> i.getPlayerID() == this.getPlayerID()).findFirst()
-                .orElse(this.CloneWithoutSubskills());
+        var playerBefore = DBManager.instance().getSpieler(previousID).stream().filter(i -> i.getPlayerID() == this.getPlayerID()).findFirst().orElse(this.CloneWithoutSubskills());
+
+        //calculate stamina subskill
+
+        //if the player has played a game
+        //lastmatch rating default value:0
+        if (m_lastMatchRating != 0) {
+            double m_iaverageStamina = DBManager.instance().getBewertungen4Player(getPlayerID())[2];
+            double m_isubStamina = (m_iaverageStamina - getLastMatchRating()) / 2;
+            m_isubStamina = Math.round(playerBefore.getStamina() - m_isubStamina);
+            if (m_isubStamina >= 1 && m_isubStamina <= 10) {
+                setStamina((int) m_isubStamina);
+            } else {
+                if (m_isubStamina < 1) {
+                    setStamina(1);
+                }
+                if (m_isubStamina > 10) {
+                    setStamina(10);
+                }
+            }
+        }
```

Comment:
i do not understand this calculation - please explain. i thought the result of substamina should be a double value in the range from 0,00 up to 0.99.